code/modules/mob/living/carbon/xenomorph/xeno_defines.dm

```diff
@@ -907,7 +909,19 @@
 		for(var/obj/item/alien_embryo/embryo in potential_host)
 			embryo.hivenumber = XENO_HIVE_FORSAKEN
 		potential_host.update_med_icon()
+	var/shipside_humans_count = 0
+	var/xenos_count = 0
+	for(var/mob/living/carbon/human/current_human as anything in GLOB.alive_human_list)
+		if(isspecieshuman(current_human) || isspeciessynth(current_human))
+			var/turf/turf = get_turf(current_human)
+			if(is_mainship_level(turf?.z))
+				shipside_humans_count++
+	for(var/mob/living/carbon/xenomorph/xenomorph as anything in totalXenos)
+		if(isfacehugger(xenomorph))
+			continue
+		xenos_count++
 	hijack_burrowed_surge = TRUE
```

Comment:
This should only be set TRUE if we will allow even a single larva to spawn. Consider not only revising the message `xeno_message(SPAN_XENOANNOUNCE("The hive swells with power! You will now steadily gain pooled larva over time."), 2, hivenumber)` but also put it here depending on if we will be getting a surge rather than just first tick hitting the surge end. If we won't get a surge just have a unique message for that or just no message at all.

Reply:
Ok, though also still consider editing that message since it still suggests the surge is indefinite. But if nothing is immediately obvious on how to better word it, its fine.

Reply:
Only thing that comes to my mind is that I can swap lines 277 and 278 in dropship_computer.dm and tell the exact number of larvae the hive will get.

Reply:
I don't think I would want it to spit out the exact count. So its fine - ultimately just wanted you to be aware how it was worded elsewhere.